### DISCUSSION TITLE
Updating Libmesh issue

### DISCUSSION THREAD

**Discussion by ahmad681** [idaholab/moose#18245] [Q&A Meshing] [answered]
Hi, 
I am having a problem with compiling my code. Last night it was working great and I was getting great results. i havent changed anything since last night. Earlier this morning, I "make clean" and tried to compile and it keeps telling me 
There appears to be a libMesh update in progress for
the branch of MOOSE you are operating on (only the
master branch contains publicly available moose-libmesh
packages).

You must provide your own libMesh either by:

    1. Uninstall moose-libmesh and build it manually via
       moose/scripts/update_and_rebuild_libmesh.sh
or
    2. Use `conda build` to build your own moose-libmesh
       package.

/home/ahmad68/projects/moose/framework/moose.mk:15: *** Build failed.  Stop.

I updated the libmesh by building my own libmesh, and still same error. Could you please help me with that? 
Thanks

**Comment by ahmad681** [comment]
Note: I asked my groupmates, they have never faced this issue before. It doesn't make sense to me yet why I would have to update my library. 
I updated it using this website https://mooseframework.inl.gov/moose/help/faq/faq_build_libmesh.html 
but it still not working. I also updated Conda, still not working too and it gives me same messege.

**Reply by GiudGiud** [reply]
Hello

Can you please first try to clean the build from the repository?
First, just try `make clobberall`. If that is not enough, commit ALL your changes, then run `git clean -xfd` (will remove everything that is not committed). 

Then we have just updated libmesh and petsc a week ago, so it s worth updating your moose and libmesh.
So
`git fetch origin; git pull origin` to get the latest moose
`conda update --all` to update libmesh.

Best,
Guillaume

**Reply by ahmad681** [reply]
Thanks for the reply Guillaume. I reached to conda update --all
 but I am getting Segmentation fault messege.

**Reply by GiudGiud** [reply]
What platform? What operating system?
Try `conda clean -a` see if it removes the segfault. Some python packages you could have pip-installed can cause seg faults for conda, but it s rare

**Reply by ahmad681** [reply]
Okay, the libmesh has now been updated successfully and when I did conda update --all, no problem popped up. But Now a problem is popping up  

It is not being able to recognize -fno-plt ! which I am not sure what this is. Thanks a lot for the help 
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/src_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/outputs_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/outputs_png_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/outputs_formatters_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/controls_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/reporters_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/timeintegrators_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/fvbcs_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/meshgenerators_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/relationshipmanagers_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/vectorpostprocessors_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/problems_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/partitioner_Unity.C...
Compiling C++ (in opt mode) /home/ahmad68/projects/moose/framework/build/unity_src/timesteppers_Unity.C...
g++: error: unrecognized command line option '-fno-plt'
g++: error: unrecognized command line option '-fno-plt'
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/src_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/outputs_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/outputs_png_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/outputs_formatters_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/controls_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/timeintegrators_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/reporters_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/fvbcs_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/meshgenerators_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/relationshipmanagers_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/partitioner_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/vectorpostprocessors_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/problems_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1
g++: error: unrecognized command line option '-fno-plt'
make: *** [/home/ahmad68/projects/moose/framework/build/unity_src/timesteppers_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1

**Reply by ahmad681** [reply]
By the way, I am using linux operating system

**Reply by GiudGiud** [reply]
did you remember to activate the moose conda envionment ? `conda activate moose`
this means a compiler flag passed is not understood by the compiler, so you are using the wrong compiller, not the one we provide with conda

**Reply by ahmad681** [reply]
Yes, I did conda activate moose

**Reply by ahmad681** [reply]
My groupmates use the same version of the code. They never had to update libmesh ever. I wonder why this happened to me also. 

Also, I am using gcc 5.2.0 I thought any 5.15 and above is fine with conda.

**Reply by GiudGiud** [reply]
5.2 should be ok.
Everyone needs to update libmesh to use the version of moose. People do that through conda updates though, so it s pretty simple. 

what does `echo $CXX; which mpicxx` return? I suspect it's set to your system compiler, not the one in conda. 

If you want to use your own system compiler instead of the one we provide in the conda environment, then you need to edit the environment variable CPP_FLAGS to remove the no-plt flag, because your (old) compiler does not support it. 
We set that flag for the conda build because we assume people are using the compilers conda installs.

**Comment by GiudGiud** [comment, marked as answer]
Any news on this?

**Reply by ahmad681** [reply]
Apparantely there was a problem with Conda installed on the cluster at my school. It seemed like it was compatible with gcc 4.8.5. 
So I had to install miniconda and mpich because for some reason openmpi was not working properly. After installing the version miniconda and mpich specified on Moose website, it worked just fine.

**Reply by GiudGiud** [reply]
Great thanks for letting us know!
I'll add a little bit to the posting guidelines to reflect this
